framework/src/st_things/things_utils/src/things_time.c

```diff
@@ -0,0 +1,43 @@
+#include <time.h>
+
+#include <things_time.h>
+
+things_result things_time_set(unsigned int second, unsigned int microsecond)
+{
+	if ((int)second < 0 || (int)microsecond < 0) {
```

Comment:
This is not correct. any unsigned value above 32k(incase of 32 bit int) will be treated as negative. Unexpected behaviour